pkg/fpga/utils.go

```diff
@@ -67,8 +67,16 @@ func checkVendorAndClass(dev commonFpgaAPI) error {
 	if err != nil {
 		return err
 	}
-	if pci.Vendor != vendorIntel || pci.Class != fpgaClass {
+
+	if pci.Class != fpgaClass {
+		return errors.Errorf("unsupported PCI class device %s  VID=%s PID=%s Class=%s", pci.BDF, pci.Vendor, pci.Device, pci.Class)
+	}
+
+	if len(os.Getenv("VID")) == 0 && pci.Vendor != vendorIntel {
+		return errors.Errorf("unsupported PCI device %s  VID=%s PID=%s Class=%s", pci.BDF, pci.Vendor, pci.Device, pci.Class)
+	} else if pci.Vendor != os.Getenv("VID") && pci.Vendor != vendorIntel {
```

Comment:
@rmr-silicom are you looking to import `pkg/fpga` in your project?

Reply:
Yes. I am adding support for the FPGA n5010 card. It uses the kernel drivers from here https://github.com/OPAE/linux-dfl-backport, but it seems the sysfs entries are mostly the same.

Reply:
@rmr-silicom can you share output of lspci for this card, to see how it is reported in terms IDs/class/etc?
Also, it would be great to see how kernel DFL drivers sysfs reports about that device.

Reply:
@mythi It has a vendor id of 0x1c2c and product ids of 0x1001, 0x1000, and the 0x12000 fpga class, of course

Reply:
Sorry, didn't catch your sysfs question... I added a few, there are a lot more...

[core@worker1 ~]$ find /sys/class/fpga_*
/sys/class/fpga_bridge
/sys/class/fpga_bridge/br0
/sys/class/fpga_manager
/sys/class/fpga_manager/fpga0
/sys/class/fpga_region
/sys/class/fpga_region/region0
/sys/class/fpga_region/region1
/sys/class/fpga_sec_mgr
/sys/class/fpga_sec_mgr/fpga_sec0

FME.0
/sys/class/fpga_region/region0/dfl-fme.0/uevent
/sys/class/fpga_region/region0/dfl-fme.0/bitstream_id
/sys/class/fpga_region/region0/dfl-fme.0/bitstream_metadata

PORT.0
/sys/class/fpga_region/region0/dfl-port.0
/sys/class/fpga_region/region0/dfl-port.0/uevent
/sys/class/fpga_region/region0/dfl-port.0/afu_id
/sys/class/fpga_region/region0/dfl-port.0/power_state
/sys/class/fpga_region/region0/dfl-port.0/ap2_event
/sys/class/fpga_region/region0/dfl-port.0/ltr

Reply:
@rmr-silicom Thanks. for PCI, I was more interested in output similar to `lspci -n -v -s XX:XX.X` where XX... is your PCI device address. Curious to see what capabilities and endpoints it reports and kernel driver.
sysfs entries look good.

Reply:
00:04.0 1200: 1c2c:1000 (rev 01)
	Physical Slot: 4
	Flags: bus master, fast devsel, latency 0
	Memory at fea00000 (64-bit, prefetchable) [size=512K]
	Memory at fe900000 (64-bit, prefetchable) [size=1M]
	Capabilities: [40] Power Management version 3
	Capabilities: [70] Express Endpoint, MSI 00
	Capabilities: [b0] MSI-X: Enable+ Count=7 Masked-
	Kernel driver in use: dfl-pci

Reply:
@rmr-silicom
> I am adding support for the FPGA n5010 card. It uses the kernel drivers from here https://github.com/OPAE/linux-dfl-backport,

Any plans to get this upstreamed?

Reply:
I believe so, but that is another development team.

Reply:
@rmr-silicom do you have on your system in `/sys/devices/pci0000:00/*/0000:04:00.0` files `subsystem_*`?

Reply:
cat /sys/devices/pci0000:00/0000:00:04.0/subsystem_vendor
0x000

cat /sys/devices/pci0000:00/0000:00:04.0/subsystem_device
0x000

What does this have to do with this patch?

Reply:
@rmr-silicom reason why I'm asking is to look at different ways on enabling devices. Getting PCI IDs from environment is one of the possible ways, but also might be possibilities to detect things either via driver or via PCI sub-properties to avoid hardcoded PCI vendor/device IDs. So, I was curious how kernel driver on your side behaves.

Reply:
Got it. Why couldn't one look for the existence of the char devices /dev/dfl*, and then from there, look in sysfs instead of using vendor id. If it is a dfl device (i hope i get terminology correct), then iterate sysfs...?

I am also using the NFD beforehand to tag the correct nodes using a deployment lspci command...

I haven't quite understood how this pkg enumerates fpga devices, yet. I just know about the PCI scan.

cat /sys/class/fpga_region/region0/device/class 
0x120000
cat /sys/class/fpga_region/region0/device/vendor 
0x1c2c
cat /sys/class/fpga_region/region0/device/device 
0x1000

Sorry, you were asking for this before..

Reply:
Yeah, we had ad-hoc with @mythi exactly about this: removing vendor check by overall, but make sure that DFL APIs are compatible still. Let us think a bit how to do that in best way...

Reply:
> removing vendor check by overall, but make sure that DFL APIs are compatible still.

@rmr-silicom can you update this PR to drop the vendor check completely and rename the function.

@kad would the existing reg expressions suffice or did you have something else in your mind?

Reply:
@rmr-silicom you can squash commits and update PR title. beside that, lgtm.

@mythi the current patch is ok from me for now. However just existing regexp for device node names might not be enough. I'd like later to implement validation of kernel driver behind the PCI device.

Reply:
@kad isn't the merge squash thing on your end? If I squash the commits, the branch changes and this request is lost.

https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges

Reply:
@rmr-silicom we don't squash commits during merge but ask the developers to do that. the ask is you do `git rebase -i HEAD~2`, combine the two commits in our branch to one and force push to remote

Reply:
Thanks for the tip!